[CI] Java 10: things are not available in profile 'compact3'

Lots of failures of the `java10-periodic` builds across all branches. For instance:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+java10-periodic/10/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.1+java10-periodic/9/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+java10-periodic/9/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+java10-periodic/10/console

Many of the failures are of the following form:

```
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:22: error: SecretKey is not available in profile 'compact3'
import javax.crypto.SecretKey;
                   ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:23: error: SecretKeyFactory is not available in profile 'compact3'
import javax.crypto.SecretKeyFactory;
                   ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:24: error: PBEKeySpec is not available in profile 'compact3'
import javax.crypto.spec.PBEKeySpec;
                        ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:109: error: SecretKeyFactory is not available in profile 'compact3'
    private final SecretKeyFactory stringFactory;
                  ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:112: error: SecretKeyFactory is not available in profile 'compact3'
    private final SecretKeyFactory fileFactory;
                  ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:135: error: SecretKeyFactory is not available in profile 'compact3'
            stringFactory = SecretKeyFactory.getInstance(stringKeyAlgo);
                            ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:136: error: SecretKeyFactory is not available in profile 'compact3'
            fileFactory = SecretKeyFactory.getInstance(fileKeyAlgo);
                          ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:311: error: PBEKeySpec is not available in profile 'compact3'
        PBEKeySpec keySpec = (PBEKeySpec) stringFactory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class);
        ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:311: error: PBEKeySpec is not available in profile 'compact3'
        PBEKeySpec keySpec = (PBEKeySpec) stringFactory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class);
                              ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:311: error: PBEKeySpec is not available in profile 'compact3'
        PBEKeySpec keySpec = (PBEKeySpec) stringFactory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class);
                                                                                                  ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:325: error: PBEKeySpec is not available in profile 'compact3'
        PBEKeySpec keySpec = (PBEKeySpec) fileFactory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class);
        ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:325: error: PBEKeySpec is not available in profile 'compact3'
        PBEKeySpec keySpec = (PBEKeySpec) fileFactory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class);
                              ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:325: error: PBEKeySpec is not available in profile 'compact3'
        PBEKeySpec keySpec = (PBEKeySpec) fileFactory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class);
                                                                                                ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:352: error: SecretKey is not available in profile 'compact3'
        SecretKey secretKey = stringFactory.generateSecret(new PBEKeySpec(value));
        ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:352: error: PBEKeySpec is not available in profile 'compact3'
        SecretKey secretKey = stringFactory.generateSecret(new PBEKeySpec(value));
                                                               ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:364: error: SecretKey is not available in profile 'compact3'
        SecretKey secretKey = stringFactory.generateSecret(new PBEKeySpec(chars));
        ^
/var/lib/jenkins/workspace/elastic+elasticsearch+5.6+java10-periodic/core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java:364: error: PBEKeySpec is not available in profile 'compact3'
        SecretKey secretKey = stringFactory.generateSecret(new PBEKeySpec(chars));
                                                               ^
```

@rjernst your name is prominent in the annotation of `KeyStoreWrapper`, could you take a look?

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -408,7 +408,11 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Adds compiler settings to the project */
     static void configureCompile(Project project) {
-        project.ext.compactProfile = 'compact3'
+        if (project.javaVersion < JavaVersion.VERSION_1_10) {
+            project.ext.compactProfile = 'compact3'
+        } else {
+            project.ext.compactProfile = 'full'
+        }
         project.afterEvaluate {
             project.tasks.withType(JavaCompile) {
                 File gradleJavaHome = Jvm.current().javaHome

